SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -86,28 +91,42 @@ template <tested_types required> auto get_tested_types() {
   }
 }
 
+auto get_all_vector_lengths() { return values_pack<1, 8, 16, 32>(); }
+
 // Run action for each of types given by named_type_pack instance
-template <template <typename, typename...> class action,
-          typename... actionArgsT, typename... types, typename... argsT>
-inline bool for_all_types(const named_type_pack<types...> &typeList,
-                          argsT &&... args) {
+template <template <typename, int, typename...> class Action, int N,
+          typename... ActionArgsT, typename... TypesT, typename... ArgsT>
```

Comment:
```suggestion
          typename... ActionArgsT, typename... Types, typename... ArgsT>
```
Minor: that's more like a pack of types, so it might be not needed to additionally state that it's a "type". What do you think?

Reply:
I've reverted this comment [here](https://github.com/intel/llvm-test-suite/pull/588/commits/b359d36a0cf7e324f477dea39bd6f3736b648c55#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R130-R136)